validator/accounts/v2/new.go

```diff
@@ -0,0 +1,265 @@
+			KeymanagerKind: keymanagerKind,
+		}
+		wallet, err = CreateWallet(ctx, walletConfig)
+		if err != nil {
+			log.Fatalf("Could not create wallet at specified path %s: %v", walletDir, err)
+		}
+		isNewWallet = true
+	}
+
+	// We initialize a new keymanager depending on the user's selected keymanager kind.
+	keymanager := initializeWalletKeymanager(ctx, wallet, isNewWallet)
+
+	// Read the new account's password from user input.
+	password := inputAccountPassword(cliCtx)
+
+	// Create a new validator account using the specified keymanager.
+	// TODO(#6220): Implement.
+	if err := keymanager.CreateAccount(ctx, password); err != nil {
+		log.Fatalf("Could not create account in wallet: %v", err)
+	}
+	return nil
+}
+
+// Initializes a keymanager. If a config file exists in the wallet, it
+// reads the config file and initializes the keymanager that way. Otherwise,
+// writes a new configuration file to the wallet and returns the initialized
+// keymanager for use.
+func initializeWalletKeymanager(ctx context.Context, wallet *Wallet, isNewWallet bool) v2keymanager.IKeymanager {
+	var keymanager v2keymanager.IKeymanager
+	var err error
+	if isNewWallet {
+		switch wallet.KeymanagerKind() {
+		case v2keymanager.Direct:
+			keymanager = direct.NewKeymanager(ctx, wallet, direct.DefaultConfig())
+		case v2keymanager.Derived:
+			log.Fatal("Derived keymanagers are unimplemented, work in progress")
+		case v2keymanager.Remote:
+			log.Fatal("Remote keymanagers are unimplemented, work in progress")
+		default:
+			log.Fatal("Keymanager type must be specified")
+		}
+		keymanagerConfig, err := keymanager.MarshalConfigFile(ctx)
+		if err != nil {
+			log.Fatalf("Could not marshal keymanager config file: %v", err)
+		}
+		if err := wallet.WriteKeymanagerConfigToDisk(ctx, keymanagerConfig); err != nil {
+			log.Fatalf("Could not write keymanager config file to disk: %v", err)
+		}
+		return keymanager
+	}
+	switch wallet.KeymanagerKind() {
+	case v2keymanager.Direct:
+		keymanager, err = direct.NewKeymanagerFromConfigFile(ctx, wallet)
+		if err != nil {
+			log.Fatal(err)
+		}
+	case v2keymanager.Derived:
+		log.Fatal("Derived keymanagers are unimplemented, work in progress")
+	case v2keymanager.Remote:
+		log.Fatal("Remote keymanagers are unimplemented, work in progress")
+	default:
+		log.Fatal("Keymanager kind must be specified")
+	}
+	return keymanager
+}
+
+// Check if a user has an existing wallet at the specified path.
+func hasWalletDir(walletPath string) (bool, error) {
+	_, err := os.Stat(walletPath)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return true, err
+}
+
+func inputWalletDir(cliCtx *cli.Context) string {
+	datadir := cliCtx.String(flags.WalletDirFlag.Name)
+	prompt := promptui.Prompt{
+		Label:    "Enter a wallet directory",
+		Validate: validateDirectoryPath,
+		Default:  datadir,
+	}
+	walletPath, err := prompt.Run()
+	if err != nil {
+		log.Fatalf("Could not determine wallet directory: %v", formatPromptError(err))
```

Comment:
maybe fatal calls should be handled by the caller of this function ? instead we can just return an error here